### PR TITLE
24 robot power down

### LINKED ISSUE #24: The user should be able to do a "power down"
Acceptance criteria: Power down makes the robot not pick new cards/ move for a round, but it will get full helath instantly after powering down. The robot does still take damage during the round from example laser fire, and can still be moved by other robots or conveyors.
  
- Handle powerdown and repair accordingly
- Make it possible to do a power down in GUI

### PR DESCRIPTION
**Pull request by runalmaas**
Fixes #24

### CODE REVIEW

**Comment by runalmaas** [comment]
Recommend trying to play one or more rounds with power down and without.
See manual for tests.

File: src/main/java/no/uib/inf112/core/player/AbstractPlayer.java [hunk 1]
```diff
@@ -25,7 +26,8 @@
 
     protected int flags;
 
-    private boolean poweredDown;
+    public boolean poweredDown;
+    public boolean willPowerDown;
```
Comment (DanielBerge) [inline src/main/java/no/uib/inf112/core/player/AbstractPlayer.java]: poweredDown and willPowerDown should be private and have getters and setters.

File: src/main/java/no/uib/inf112/core/player/Player.java [hunk 2]
```diff
@@ -31,8 +31,12 @@ public Player(int x, int y, @NotNull Direction direction, MapHandler map) {
                     return;
 
                 }
-                setPoweredDown(!isPoweredDown());
-                System.out.println("Powered down? " + isPoweredDown());
+
+                if(poweredDown){
+                    poweredDown = false;
+                }
+                willPowerDown = !willPowerDown;
```
Comment (DanielBerge) [inline src/main/java/no/uib/inf112/core/player/Player.java]: Use the willPowerDown() method

File: src/test/java/no/uib/inf112/core/player/RobotTest.java [hunk 3]
```diff
@@ -196,6 +200,28 @@ public Vector2Int getBackup() {
             return new Vector2Int(0, 0);
         }
 
+        @Override
+        public void clean(@NotNull Tile tile) {
+            //EMPTY
```
Comment (DanielBerge) [inline src/test/java/no/uib/inf112/core/player/RobotTest.java]: Add a more describing comment

File: src/test/java/no/uib/inf112/core/player/PlayerTest.java [hunk 4]
```diff
@@ -192,5 +197,27 @@ public void playerConstructorSucceedForAllValidPos() {
         public ComparableTuple<Card, IPlayer> getNextCard(int id) {
             return null;
         }
+
+        @Override
+        public void clean(@NotNull Tile tile) {
+            //EMPTY
```
Comment (DanielBerge) [inline src/test/java/no/uib/inf112/core/player/PlayerTest.java]: Add a more describing comment

File: src/test/java/no/uib/inf112/core/map/tile/tiles/WallTileTest.java [hunk 5]
```diff
@@ -141,6 +145,22 @@ public Vector2Int getBackup() {
...
+        }
+
+        @Override
+        public boolean canDoAction(@NotNull Tile tile) {
+            return false;
+        }
+
+        @Override
+        public void clean(@NotNull Tile tile) {
+            //EMPTY
```
Comment (DanielBerge) [inline src/test/java/no/uib/inf112/core/map/tile/tiles/WallTileTest.java]: Add a more describing comment

**Pull request merged**